include/openthread/link.h

```diff
@@ -1004,6 +1005,72 @@ bool otLinkIsPromiscuous(otInstance *aInstance);
  */
 otError otLinkSetPromiscuous(otInstance *aInstance, bool aPromiscuous);
 
+/**
+ * This function gets the CSL channel.
+ *
+ * @param[in]  aInstance      A pointer to an OpenThread instance.
+ *
+ * @returns The CSL channel.
+ *
+ */
+uint8_t otLinkCslGetChannel(otInstance *aInstance);
+
+/**
+ * This function sets the CSL channel.
+ *
+ * @param[in]  aInstance      A pointer to an OpenThread instance.
+ * @param[in]  aChannel       The CSL sample channel.
+ *
+ * @retval OT_ERROR_NONE           Successfully set the CSL parameters.
+ * @retval OT_ERROR_INVALID_ARGS   Invalid @p aChannel.
+ *
+ */
+otError otLinkCslSetChannel(otInstance *aInstance, uint8_t aChannel);
```

Comment:
Here is my understanding:
Section 3.2.6.3.1
>  CSL unsynchronized transmission and reception are outside the scope of the Thread specification.

This means `Thread 1.2` doesn't specify how should the CSL receiver handle unsynchronized CSL transmissions.

And Section 4.6.5.1.2 means that, if the application expects the CSL Receiver to process unsynchronized CSL transmissions, the Receiver should use a different channel. Then the next two paragraph said:
> During the attachment phase, the SSED Child MAY include the CslChannel TLV. If included,
> the Parent MUST perform the CSL transmissions on the channel indicated by the Child in the
> CslChannel TLV. In case of a successful CSL transmission, the SSED MUST send the IEEE
> 802.15.4 ACK on the channel indicated in the CslChannel TLV.
> If no CslChannel TLV was included by the Child during attachment, the Parent MUST
> perform CSL transmissions on the Thread Network channel.
So I think we should enable CSL Receiver to receive on another channel. Besides, in test harness, a case is certifying that.

Reply:
Is there a certification test for supporting a different CSL channel? If not, I am fine to drop support for this until we have a strong use case for it.

Reply:
Yes, there's a certification test to validate that a Router can support a SSED that modifies the CSL Channel TLV. Currently in low power test plan r07, case 5_3_10.

Reply:
Fine to me to keep it for certification's consideration for now.
But if `CSL unsynchronized transmission and reception are outside the scope of the Thread specification`, should we propose to Thread Group to unify the wording in spec or tests in cert spec to avoid a difference CSL channel?
@jwhui @Irving-cl